File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    lost_s = set(lost)
    reserve_d = set(reserve)
    for l in lost:
        if l in reserve_d:
            lost_s.remove(l)
            reserve_d.remove(l)
    
    for i in range(1, n + 1):
        if not i in lost_s:
            answer += 1
            continue
            
        if i in lost_s:
            front = i - 1
            back = i + 1
            if front in reserve_d:
                answer += 1
                reserve_d.remove(front)
                continue
                
            if back in reserve_d:
                answer += 1
                reserve_d.remove(back)
                continue
                
    return answer

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;
        Set<Integer> originalLost = new HashSet<>();
        Set<Integer> originalReserve = new HashSet<>();
        
        for(int l : lost) originalLost.add(l);
        for(int r : reserve) originalReserve.add(r);
        
        List<Integer> dup = new ArrayList<>();
        
        for(Integer l : originalLost) if(originalReserve.contains(l)) dup.add(l);
        
        for(Integer d : dup) {
            originalLost.remove(d);
            originalReserve.remove(d);
        }
        
        for(Integer l : originalLost) {
            if(originalReserve.contains(l - 1)) {
                answer += 1;
                originalReserve.remove(l - 1);
            } else if(originalReserve.contains(l + 1)) {
                answer += 1;
                originalReserve.remove(l + 1);
            }
        }
        
        return n - (lost.length - dup.size() - answer);
    }
}